core/lib/src/data/bytes.rs

```diff
@@ -0,0 +1,88 @@
+use std::borrow::Cow;
+use std::io;
+
+use crate::data::{Capped, FromData, Limits, Outcome, N};
+use crate::form::{DataField, Errors, FromFormField, ValueField};
+use crate::http::{ContentType, Status};
+use crate::{fs::FileName, outcome::IntoOutcome, Data, Request};
+
+#[derive(Debug)]
+pub struct Bytes<'v> {
+    pub file_name: Option<&'v FileName>,
+    pub content_type: Option<ContentType>,
+    pub content: Cow<'v, [u8]>,
```

Comment:
Just wondering, is there a reason for using `Cow<[u8]>` here instead of `Vec<u8>`? As far as I can tell converting a `Cow<[T]>` back into a `Vec<T>` would cause a reallocation of the entire file even if it's owned and I would rather have the `Vec` for my use-case.

Reply:
I'm finally done with having to work so much, so hopefully I'll be more responsive now.
On line 52, I use it for the borrowed variant, so there is a purpose in using it and there should not be an unnecessary (re)allocation because:
> [Calling into_owned on a Cow::Owned returns the owned data. The data is moved out of the Cow without being cloned.](https://doc.rust-lang.org/std/borrow/enum.Cow.html#method.into_owned)

For extra clarification, the type of `content` doesn't mention `Vec` because it is inferred from the [ToOwned implementation for slices](https://doc.rust-lang.org/src/alloc/slice.rs.html#782) that specifies its `Owned` variant is `Vec`.